### PR TITLE
Swap modifier

### LINKED ISSUE #46: Add swap modifier (.s) to all instructions that need it
In particular, all ptr instructions ,` ptr.add , ptr.sub, ptr.shrink and ptr.pack` should use it, but when looking at the opcode table, there is not any entrance for this opcodes where the swap_flag is true, this is noted on the [Notion questions](https://www.notion.so/lambdaclass/Era-VM-Docs-and-Questions-6d6c00a49e1744b8a6f8f171f07a4a7e).

### PR DESCRIPTION
**Pull request by juan518munoz**
Closes #46
Implement `.s` modifier.

### CODE REVIEW

File: src/op_handlers/div.rs [hunk 1]
```diff
@@ -3,7 +3,9 @@ use crate::value::TaggedValue;
 use crate::{opcode::Opcode, state::VMState};
 
 pub fn _div(vm: &mut VMState, opcode: &Opcode) {
+    println!("swap flag: {}", opcode.swap_flag);
```
Comment (fkrause98) [inline src/op_handlers/div.rs:6]: Remove these prints.

File: src/address_operands.rs [hunk 2]
```diff
@@ -74,6 +74,11 @@ pub fn address_operands_read(vm: &mut VMState, opcode: &Opcode) -> (TaggedValue,
                 }
             }
         }
+    };
+    if opcode.swap_flag {
```
Comment (fkrause98) [inline src/address_operands.rs:78]: Could there be a combination of inputs that when swapped becomes non-valid? Eg: If I do `add.swap 1, r0, r1`

**Review by fkrause98 — CHANGES_REQUESTED** [review]

File: src/address_operands.rs [hunk 3]
```diff
@@ -74,6 +74,11 @@ pub fn address_operands_read(vm: &mut VMState, opcode: &Opcode) -> (TaggedValue,
...
+        (op2, op1)
+    } else {
+        (op1, op2)
```
Comment (gianbelinche) [inline src/address_operands.rs:81]: I'm not sure if this is how we should do it, I think not all opcodes support swapping, maybe this should be implemented in each opcode. (Please check this, maybe its fine)

**Review by gianbelinche — CHANGES_REQUESTED** [review]

File: src/address_operands.rs (hunk 2 above)
Comment (juan518munoz) [reply src/address_operands.rs:78]: Compiler forbids this kind of instructions:
```asm
	div.s r1, 3, r3, r4
```
throws:
```bash
error: assembly parsing: can not parse text section element: cannot parse lines: {6: ("div.s r1, 3, r3, r4", InvalidOperandImmInRegLocation("3"))}
```

File: src/address_operands.rs (hunk 3 above)
Comment (juan518munoz) [reply src/address_operands.rs:81]: As you say, some functions, like `Mul`, don't support the `.s`, this is taken care of by the compiler, which forbids bulding this kind of code:
```asm
	mul.s 3, r1, r3, r4
```
throws:
```bash
 error: assembly parsing: can not parse text section element: cannot parse lines: {6: ("mul.s 3, r1, r3, r4", UnknownArgument("Mul instruction contains unknown modifiers: {\"s\"}"))}
 ```

**Review by gianbelinche — APPROVED** [review]

**Comment by gianbelinche** [comment]
LGTM!

**Review by fkrause98 — APPROVED** [review]
Looks good. Feel free to merge after solving the merge conflict.

**Pull request merged**